pymc3/step_methods/hmc/quadpotential.py

```diff
@@ -29,6 +34,8 @@ def quad_potential(C, is_cov, as_cov):
     -------
     q : Quadpotential
     """
+    if isquadpotential(C):
+        return C
```

Comment:
This is also a good feature regardless of the stuff below (for the separate PR).